src/benchmarks/micro/coreclr/perflab/DictinoaryExpansion.cs

```diff
@@ -0,0 +1,158 @@
+        public Type FuncOnGenClass(int level)
+        {
+            switch (level)
+            {
+                case 0: return typeof(T);
+                case 1: return typeof(List<T>);
+                case 2: return typeof(List<List<T>>);
+                case 3: return typeof(List<List<List<T>>>);
+                case 4: return typeof(List<List<List<List<T>>>>);
+                case 5: return typeof(List<List<List<List<List<T>>>>>);
+                case 6: return typeof(List<List<List<List<List<List<T>>>>>>);
+                case 7: return typeof(List<List<List<List<List<List<List<T>>>>>>>);
+                case 8: return typeof(List<List<List<List<List<List<List<List<T>>>>>>>>);
+                case 9: return typeof(List<List<List<List<List<List<List<List<List<T>>>>>>>>>);
+                case 10: return typeof(List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>);
+                case 11: return typeof(List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>);
+                case 12: return typeof(List<List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>>);
+                default: return typeof(List<List<List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>>>);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public Type FuncOnGenClass2(int level)
+        {
+            switch (level)
+            {
+                case 0: return typeof(T);
+                case 1: return typeof(List<T>);
+                case 2: return typeof(List<List<T>>);
+                case 3: return typeof(List<List<List<T>>>);
+                case 4: return typeof(List<List<List<List<T>>>>);
+                case 5: return typeof(List<List<List<List<List<T>>>>>);
+                case 6: return typeof(List<List<List<List<List<List<T>>>>>>);
+                case 7: return typeof(List<List<List<List<List<List<List<T>>>>>>>);
+                case 8: return typeof(List<List<List<List<List<List<List<List<T>>>>>>>>);
+                case 9: return typeof(List<List<List<List<List<List<List<List<List<T>>>>>>>>>);
+                case 10: return typeof(List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>);
+                case 11: return typeof(List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>);
+                case 12: return typeof(List<List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>>);
+                default: return typeof(List<List<List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>>>);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void FuncOnGenClassTest()
+        {
+            var o1 = new GenClass<string>();
+            var o2 = new GenClass<object>();
+            var o3 = new GenClass<DictionaryExpansion>();
+
+            for (int j = 0; j < DictionaryExpansion.s_Iterations; j++)
```

Comment:
You should not need to add a loop here, BenchmarkDotNet scales the number of inner iterations using a dynamic heuristic based on a `IterationTime` setting:

https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#Loops